app/models/machine.rb

```diff
@@ -1,5 +1,10 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
+  has_many :snacks
   belongs_to :owner
+
+  def average_snack_price
+    self.snacks.average(:price)
+  end
```

Comment:
Good job on this method. You don't necessarily need the self in the method, but it's not harmful to have it there either.